source/Halibut/Util/Try.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System;
+
+namespace Halibut.Util
+{
+    public static class Try
+    {
+        public static void CatchingError(Action tryThisAction, Action<Exception> onFailure)
```

Comment:
Nit: Instead of passing a required lambda, this could be nullable (even with a default of null), so we don't have to pass in a dummy lambda (like we do in `PollingClient.Dispose`)